src/main/java/fun/epak/pak/service/UserService.java

```diff
@@ -100,9 +103,24 @@ public void saveChangedUser(ChangeUserDataRequest userData, MultipartFile multip
         userRepository.save(user);
     }
 
-    public OtherUserProfileData loadOtherUserProfileData(long id) {
+    public OtherUserProfileData loadOtherUserProfileData(String email, long id) {
         User user = userRepository.findById(id).orElseThrow();
+        User viewer = userRepository.findByEmail(email).orElseThrow();
         String userImagePath = imageBaseAddress + user.getId() + "/" + user.getImageName();
-        return OtherUserProfileData.of(user, userImagePath);
+        return OtherUserProfileData.of(user, userImagePath, viewer);
+    }
+
+    public void changeUserSubscriptionList(String email, long id) {
+        User user = userRepository.findByEmail(email).orElseThrow();
+        User subscriptedUser = userRepository.findById(id).orElseThrow();
+        if (user.equals(subscriptedUser)) {
+            throw new SubscribeYourselfException("You cant subscribe to yourself");
+        }
+        if (user.getSubscriptions().contains(subscriptedUser)) {
+            user.getSubscriptions().remove(subscriptedUser);
```

Comment:
Dlaczego? Za to powinien odpowiadać osobny endpoint, albo w requeście powinna iść informacja czy odsubskrybujemy czy subskrybujemy.

Reply:
Może podam przykład. Jeśli walniemy z 2 dwóch okien przycisk odsubuj to w efekcie zasubuje na nowo :)